examples/keccak.rs

```diff
@@ -0,0 +1,449 @@
+use circom_scotia::r1cs::CircomInput;
+use pasta_curves::pallas;
+use std::fmt::Debug;
+use std::marker::PhantomData;
+use std::sync::Arc;
+use std::time::Instant;
+use tiny_keccak::{Hasher, Keccak};
+
+use lurk::circuit::gadgets::circom::{CircomGadget, CircomGadgetReference};
+use lurk::circuit::gadgets::pointer::AllocatedPtr;
+
+#[cfg(not(target_arch = "wasm32"))]
+use lurk::coprocessor::circom::non_wasm::CircomCoprocessor;
+
+use lurk::coprocessor::gadgets::{chain_car_cdr, construct_list};
+use lurk::coprocessor::{CoCircuit, Coprocessor};
+use lurk::dual_channel::dummy_terminal;
+use lurk::field::LurkField;
+use lurk::lang::Lang;
+use lurk::lem::eval::{
+    evaluate, make_cprocs_funcs_from_lang, make_eval_step_from_config, EvalConfig,
+};
+use lurk::lem::{pointers::Ptr, store::Store};
+use lurk::proof::supernova::SuperNovaProver;
+use lurk::proof::RecursiveSNARKTrait;
+use lurk::public_parameters::{instance::Instance, supernova_public_params};
+use lurk::state::user_sym;
+use lurk::tag::{ExprTag, Tag};
+use lurk_macros::Coproc;
+use serde::{Deserialize, Serialize};
+
+const REDUCTION_COUNT: usize = 10;
+
+/*************************************************
+ * Utilities
+ *************************************************/
+fn string_to_le_bits(input_string: &str) -> Vec<u32> {
+    let bytes = input_string
+        .chars()
+        .flat_map(|c| c.to_string().into_bytes())
+        .collect::<Vec<_>>();
+
+    bytes_to_bits(&bytes)
+        .iter()
+        .map(|b| if *b { 1 } else { 0 })
+        .collect::<Vec<_>>()
+}
+
+// Transforms a slice of bits in a slice of bytes. Fills the bytes from least to most significant
+// bit.
+fn bits_to_bytes(bits: &[bool]) -> Vec<u8> {
+    let mut bytes = vec![0; (bits.len() + 7) / 8]; // Initialize a vector with zeroes
+
+    for (i, &bit) in bits.iter().enumerate() {
+        // Iterate over each bit with its index
+        let byte_index = i / 8; // Calculate the byte index for the current bit
+        if bit {
+            // If the current bit is true,
+            bytes[byte_index] |= 1 << (i % 8); // Set the corresponding bit in the byte
+        }
+    }
+    bytes // Return the array of bytes
+}
+
+// Transforms a slice of bytes to a slice of bits. When dividing one byte in bits, order the bits
+// from the least significant to the most significant one.
+fn bytes_to_bits(bytes: &[u8]) -> Vec<bool> {
+    let mut bits = Vec::new(); // Create a new, empty vector to store bits
+
+    for &byte in bytes.iter() {
+        // Iterate over each byte in the input slice
+        for j in 0..8 {
+            // For each bit in the byte
+            if byte & (1 << j) > 0 {
+                // Check if the bit is set
+                bits.push(true); // If the bit is set, push 1 to the vector
+            } else {
+                bits.push(false); // If the bit is not set, push 0
+            }
+        }
+    }
+    bits // Return the vector of bits
+}
```

Comment:
Could we just import https://github.com/lurk-lab/bellpepper/blob/02cdf17607a4a31ea894e6b0fe9b21e6e6bc9331/crates/bellpepper/src/gadgets/multipack.rs#L40-L45?

Reply:
Switched to `bytes_to_bit_le`